### PR TITLE
Global Aggregate Window operator for Franz streaming

### PR DESCRIPTION
**Pull request by ameyc**
This issue adds

1. Continuous package to `physical-plan`
2. Adds a global aggregate operator for Streaming Windows. We can now do things like `min`, `max`, `topK` etc to get columnar aggregates.
3. Adds a concept of StreamingWindow to the logical plan.
4. Adds a physical optimizer rule to Coalesce partitions before feeding them to the window aggregate operator.

### CODE REVIEW

File: datafusion/core/src/physical_optimizer/pipeline_checker.rs [hunk 1]
```diff
@@ -69,6 +69,7 @@ pub fn check_finiteness_requirements(
     input: Arc<dyn ExecutionPlan>,
     optimizer_options: &OptimizerOptions,
 ) -> Result<Transformed<Arc<dyn ExecutionPlan>>> {
+    println!("{:?}", input);
```
Comment (emgeee) [inline datafusion/core/src/physical_optimizer/pipeline_checker.rs]: extra print?

File: datafusion/expr/src/logical_plan/plan.rs [hunk 2]
```diff
@@ -158,6 +158,8 @@ pub enum LogicalPlan {
     Unnest(Unnest),
     /// A variadic query (e.g. "Recursive CTEs")
     RecursiveQuery(RecursiveQuery),
+
```
Comment (emgeee) [inline datafusion/expr/src/logical_plan/plan.rs]: think we need a comment here

**Review by emgeee — APPROVED** [review]
Added a few tiny nits but overall I expect we'll test this as we develop more

File: datafusion/core/src/physical_optimizer/pipeline_checker.rs (hunk 1 above)
Comment (ameyc) [reply datafusion/core/src/physical_optimizer/pipeline_checker.rs]: was trying to trace that. ty.

**Pull request merged**